src/create-select.tsx

```diff
@@ -6,35 +6,32 @@ import {
   on,
 } from "solid-js";
 
-type Option = any;
+type Value<V> = V | V[];
 
-type SingleValue = any;
 
-type Value = SingleValue | SingleValue[];
```

Comment:
This here is the raison d'être; boo hiss `any`. Now this yields an error:

```tsx
<Select initialValue={1} options={["a", "b"]} optionToValue={(o) => { return new Date()}} />
// Type '(o: string) => Date' is not assignable to type '(option: string) => number'.
//  Type 'Date' is not assignable to type 'number'.
```

You can also do explicit types like

```tsx
<Select<string, string> initialValue={1} options={["a", "b"]} />
// Type 'number' is not assignable to type 'Value<string> | undefined'.
```